src/Microsoft.ML.Dnn/LearningRateScheduler.cs

```diff
@@ -0,0 +1,156 @@
+
+        /// <summary>
+        /// Linear Scale rule and LR Decay construtor assigns a default LR scheduler.
+        /// </summary>
+        public LsrDecay(float baseLearningRate = 0.1f)
+        {
+            LrSchedule = new float[,] { { 182, 0.001f }, { 136, 0.01f }, { 91, 0.1f }, { 0, 1.0f } };
+            BaseLearningRate = baseLearningRate;
+        }
+
+        /// <summary>
+        /// Linear Scale rule and LR Decay construtor assigns a user defined LR scheduler.
+        /// </summary>
+        public LsrDecay(float[,] lrschedule, float baseLearningRate = 0.1f)
+        {
+            LrSchedule = lrschedule;
+            BaseLearningRate = baseLearningRate;
+        }
+
+        /// <summary>
+        /// This function returns the corresponding scaling factor or multiplier for the given epoch from the LrSchedule.
+        /// </summary>
+        private float GetLearningRateScheduleMultiplier(int epoch)
+        {
+            for(int i = 0; i < LrSchedule.Length; i++)
+            {
+                if (epoch >= LrSchedule[i,0])
+                {
+                    return LrSchedule[i, 1];
+                }
+            }
+            return 1.0f;
+        }
+
+        /// <summary>
+        /// This function returns the Learning rate using linear scale rule and LR decay.
+        /// </summary>
+        internal override float GetLearningRate(TrainState trainstate)
+        {
+            float learningrate;
+            float initialLearningRate = BaseLearningRate * trainstate.BatchSize / 128;
+            learningrate = initialLearningRate * GetLearningRateScheduleMultiplier(trainstate.CurrentEpoch);
+            return learningrate;
+        }
+
+    }
+
+    /// <summary>
+    /// This class implements Exponential Learning rate decay.
+    /// Implemented from the tensorflow documentation.
+    /// Source: https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/exponential_decay
+    /// Default values and implementation of learning rate is from Tensorflow Slim model tests.
+    /// Source : https://github.com/tensorflow/models/blob/master/research/slim/train_image_classifier.py
+    /// </summary>
+    public sealed class ExponentialLRDecay : LearningRateScheduler
+    {
+        /// <summary>
+        /// Initial learning rate.
+        /// </summary>
+        public float LearningRate;
+
+        /// <summary>
+        /// The number of batches seen by the graph so far.
+        /// </summary>
+        public int GlobalStep;
+
+        /// <summary>
+        /// Number of decay steps
+        /// </summary>
+        public int DecaySteps;
+
+        /// <summary>
+        /// Learning rate decay factor.
+        /// </summary>
+        public float DecayRate;
+
+        /// <summary>
+        /// If Staircase is True the learning rate decays at discrete intervals and the decayed learning rate follows a staircase function.
+        /// </summary>
+        public bool Staircase { get; }
```

Comment:
Why is this a property, but everything else in this class is a field?

IMO - we should only be exposing properties publicly in APIs.

Reply:
Thanks changed it. It doesn't have to be a property. I added that by mistake.